crates/amaru-consensus/src/consensus/header_validation.rs

```diff
@@ -47,42 +43,34 @@ const EVENT_TARGET: &str = "amaru::consensus";
 )]
 pub fn assert_header<'a>(
     point: &Point,
-    header: &'a babbage::MintedHeader<'a>,
-    epoch_to_nonce: &HashMap<Epoch, Hash<32>>,
+    header: &'a MintedHeader<'a>,
+    epoch_nonce: &Nonce,
     ledger: &dyn HasStakeDistribution,
 ) -> Result<(), ConsensusError> {
-    let epoch = epoch_from_slot(header.header_body.slot);
-
-    if let Some(epoch_nonce) = epoch_to_nonce.get(&epoch) {
-        let active_slot_coeff: FixedDecimal =
-            FixedDecimal::from(1_u64) / FixedDecimal::from(ACTIVE_SLOT_COEFF_INVERSE as u64);
-
-        praos::header::assert_all(header, ledger, *epoch_nonce, &active_slot_coeff)
-            .and_then(|assertions| {
-                use rayon::prelude::*;
-                assertions.into_par_iter().try_for_each(|assert| assert())
-            })
-            .map_err(|e| ConsensusError::InvalidHeader(point.clone(), e))
-    } else {
-        Ok(())
-    }
+    let active_slot_coeff: FixedDecimal =
+        FixedDecimal::from(1_u64) / FixedDecimal::from(ACTIVE_SLOT_COEFF_INVERSE as u64);
+
+    praos::header::assert_all(header, ledger, epoch_nonce, &active_slot_coeff)
+        .and_then(|assertions| {
+            use rayon::prelude::*;
+            assertions.into_par_iter().try_for_each(|assert| assert())
+        })
+        .map_err(|e| ConsensusError::InvalidHeader(point.clone(), e))
 }
 
 pub struct Consensus {
     peer_sessions: HashMap<Peer, PeerSession>,
-    chain_selector: Arc<Mutex<ChainSelector<ConwayHeader>>>,
+    chain_selector: Arc<Mutex<ChainSelector<Header>>>,
     ledger: Box<dyn HasStakeDistribution>,
-    store: Arc<Mutex<dyn ChainStore<ConwayHeader>>>,
-    epoch_to_nonce: HashMap<Epoch, Hash<32>>,
+    store: Arc<Mutex<dyn ChainStore<Header>>>,
 }
 
 impl Consensus {
     pub fn new(
         peer_sessions: Vec<PeerSession>,
         ledger: Box<dyn HasStakeDistribution>,
-        store: Arc<Mutex<dyn ChainStore<ConwayHeader>>>,
-        chain_selector: Arc<Mutex<ChainSelector<ConwayHeader>>>,
-        epoch_to_nonce: HashMap<Epoch, Hash<32>>,
+        store: Arc<Mutex<dyn ChainStore<Header>>>,
```

Comment:
Nice we have this dependency removed 🎉

Reply:
Well, don't get excited 😅  ... since I renamed `Header` to `IsHeader`, this `Header` actually still refers to a `ConwayHeader`! So, nothing has changed really.